src/dune_rules/gen_rules.ml

```diff
@@ -373,38 +373,45 @@ let has_rules subdirs f =
        ; rules
        })
 
-let under_melange_emit_target ~dir =
-  let parent = Path.Build.parent_exn dir in
-  let+ stanzas = Only_packages.stanzas_in_dir parent in
-  match stanzas with
-  | None -> None
-  | Some stanzas ->
-    List.find_map stanzas.stanzas ~f:(function
-      | Melange_stanzas.Emit.T mel ->
-        let target_dir = Path.Build.relative parent mel.target in
-        Option.some_if (Path.Build.equal target_dir dir) mel
-      | _ -> None)
+let rec under_melange_emit_target ~dir =
+  match Path.Build.parent dir with
```

Comment:
Why does the function start looking on the parent folder? Won't it miss the case where there is only 1 level `(subdir <emit_target> ...)`?

Reply:
I think we have a test for this case and it works.

We are always looking for the parent folder because we only care about detecting whether we're under `$stanza_dir/$target`. So we always need to take at least one parent to get up to `$stanza_dir`